scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/VariableSubstitutor.java

```diff
@@ -89,16 +93,37 @@ public static String filterAndUpdate(String input,
         }
 
         String output = input;
-        for (Map.Entry<String, String> replacement : buildSubstitutes(variables).entrySet()) {
-            if (replacement.getValue() != null) {
-                output = output.replace(replacement.getKey(), replacement.getValue());
-            }
-        }
+        Map<String, String> substitutes = buildSubstitutes(variables);
+        output = replaceRecursively(output, substitutes);
 
         return output;
     }
 
     /**
+     * Replace the given string with a list of substitutions recursively. Recursion will be limited to MAXIMUM_DEPTH.
+     *
+     * @param value       string used to apply replacement
+     * @param substitutes map of substitutions
+     * @return a new string where all replacements were performed
+     */
+    private static String replaceRecursively(String value, Map<String, String> substitutes) {
+        boolean anyReplacement;
+        int depthCount = 0;
+        do {
+            anyReplacement = false;
+            depthCount++;
+            for (Map.Entry<String, String> replacement : substitutes.entrySet()) {
+                if (replacement.getValue() != null) {
+                    String newOutput = value.replace(replacement.getKey(), replacement.getValue());
+                    anyReplacement = anyReplacement || !newOutput.equals(value);
+                    value = newOutput;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "value". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1226)